### PR TITLE
Add test cases to cover failure scenarios of `bal persist generate` 

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose

Fixes: [3840](https://github.com/ballerina-platform/ballerina-standard-library/issues/3840)

## Checklist
- [x] Linked to an issue
- [ ] ~Updated the specification~
- [ ] ~Updated the changelog~
- [x] Added tests

### CODE REVIEW

File: persist-cli-tests/src/test/java/io/ballerina/persist/tools/ToolingGenerateTest.java [hunk 1]
```diff
@@ -68,10 +68,10 @@ public void testGenerateAllEntityFieldTypes() {
         assertGeneratedSources("tool_test_generate_6", GENERATE);
     }
 
-    @Test(enabled = true) // this testcase not valid.
-    @Description("Use case where a entity is located inside a module")
-    public void testGenerateClientWithEntityInModule() {
-        assertGeneratedSources("tool_test_generate_7", GENERATE);
+    @Test(enabled = true) //disabled sue to compiler plugin changes
```
Comment (daneshk) [inline persist-cli-tests/src/test/java/io/ballerina/persist/tools/ToolingGenerateTest.java:71]:
```suggestion
    @Test(enabled = true)
```

Comment (daneshk) [reply persist-cli-tests/src/test/java/io/ballerina/persist/tools/ToolingGenerateTest.java:71]: Do we need this commend now

**Review by daneshk — COMMENTED** [review]
Add minor comment

File: persist-cli-tests/src/test/java/io/ballerina/persist/tools/ToolingGenerateTest.java (hunk 1 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/java/io/ballerina/persist/tools/ToolingGenerateTest.java:71]: No we don't it was added before updating to the latest compiler plugin

File: persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java [hunk 2]
```diff
@@ -80,11 +82,10 @@ public static BuildProject getBuildProject(Path projectPath) throws BalException
             errorMessage.append("error occurred when validating the project. ");
             int validErrors = 0;
             for (Diagnostic diagnostic : diagnosticResult.errors()) {
+                String[] path = diagnostic.location().lineRange().filePath().split(FILE_PATH_SEPARATOR);
+                String fileName = path[path.length - 1].trim();
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java:86]:
```suggestion
                Path path = Paths.get(diagnostic.location().lineRange().filePath());
                String fileName = path.getFileName();
```

File: persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java [hunk 3]
```diff
@@ -80,11 +82,10 @@ public static BuildProject getBuildProject(Path projectPath) throws BalException
...
 
-                if (!diagnostic.diagnosticInfo().code().equals(DiagnosticErrorCode
-                        .INCOMPATIBLE_TYPES.diagnosticId()) &&
-                        !diagnostic.diagnosticInfo().code().equals(DiagnosticErrorCode
-                        .INCOMPATIBLE_MAPPING_CONSTRUCTOR.diagnosticId())) {
+                if (!Files.exists(projectPath.toAbsolutePath().resolve(GENERATED_SOURCE_DIRECTORY).resolve(fileName))) {
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java:88]: This file may exist inside the module directory inside the generated directory.

Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java]: Updated in [5f71463](https://github.com/ballerina-platform/persist-tools/pull/83/commits/5f714630b23b5693dfafd59c950713a508753f05)

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxConstants.java [hunk 4]
```diff
@@ -235,5 +237,6 @@ private InheritedTypeReferenceConstants() {
 
 
     public static final String MYSQL_DRIVER = "mysql.driver";
+    public static final String FILE_PATH_SEPARATOR = Pattern.quote(System.getProperty("file.separator"));
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxConstants.java:240]: Do we need this constant.

Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxConstants.java]: removed in [fb621ad](https://github.com/ballerina-platform/persist-tools/pull/83/commits/fb621ad4f4d1fb538bf618cef7dfaf37436c5292)

**Review by daneshk — APPROVED** [review]

**Pull request merged**